inc/authldap.class.php

```diff
@@ -1763,9 +1763,8 @@ static function getAllUsers(array $options, &$results, &$limitexceeded) {
          } else {
             //Ldap synchronisation : look if the user exists in the directory
             //and compares the modifications dates (ldap and glpi db)
-            $userfound = false;
-            if (!empty($ldap_users[$user[$field_for_db]])
-                || ($userfound = self::dnExistsInLdap($user_infos, $user['user_dn']))) {
+            $userfound = self::dnExistsInLdap($user_infos, $user['user_dn']);
+            if (!empty($ldap_users[$user[$field_for_db]])) {
```

Comment:
not approved finally, should we not have the part `|| $userfound` in the `if` block.
We have a `if` at 1770 for this var but also a `else` line 1784

Reply:
Since `$userfound` is tested just below; it is not relevant in the first one.

Reply:
Yes, you're right; fixed.